airflow/sensors/external_task.py

```diff
@@ -318,3 +325,19 @@ def get_serialized_fields(cls):
         if not cls.__serialized_fields:
             cls.__serialized_fields = frozenset(super().get_serialized_fields() | {"recursion_depth"})
         return cls.__serialized_fields
+
+
+@attr.s(auto_attribs=True)
```

Comment:
Why is `attr` needed here?

Reply:
@uranusjr Just for deprecation warning, otherwise it not call

https://github.com/apache/airflow/blob/03389b9c7cfd48bce1e35c49ed4f0f2230e5f473/airflow/sensors/external_task.py#L337-L343


I've also tried to use initially just regular `__init__` method but for some reason Deprecation warning wouldn't show and I couldn't find any additional filter which suppress  `DeprecationWarning`
```python
class ExternalTaskSensorLink(ExternalDagLink):
    """
    This external link is deprecated.
    Please use :class:`airflow.sensors.external_task.ExternalDagLink`.
    """

    def __init__(self, *args, **kwargs):
        warnings.warn(
            "This external link is deprecated. "
            "Please use :class:`airflow.sensors.external_task.ExternalDagLink`.",
            DeprecationWarning,
            stacklevel=2,
        )
        super.__init__(*args, **kwargs)
```

Probably for this point of time DepricationWarning is redundant.
I could just assign `ExternalTaskSensorLink` to `ExternalTaskMarker`

Main motivation why I rename link and mark current as deprecated because in theory both `ExternalTaskSensor` and `ExternalTaskMarker should have two different links: 
1. To External DAG (always)
2. To External Task Instance

Reply:
`attrs` probably does something funky to the parent class that makes regular subclassing mechanisms not work correctly. This is fine.